presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/FunctionCallRewriter.java

```diff
@@ -475,9 +517,18 @@ private static Expression parseOriginalFunctionCall(String functionCallSpec)
                         return;
                     }
                 }
-                throw new IllegalArgumentException(String.format("Argument of type %s from %s is not supported.", argument.getClass().getSimpleName(), functionCallSpec));
+                throw new IllegalArgumentException(String.format("Argument of type %s from %s is not supported.", argument.getClass().getSimpleName(), expressionSpec));
             });
         }
+        else if (expression instanceof Cast) {
+            Cast cast = (Cast) expression;
+            if (!(cast.getExpression() instanceof Identifier)) {
+                throw new IllegalArgumentException(String.format("Argument of type %s from %s is not supported.", cast.getExpression().getClass().getSimpleName(), expressionSpec));
```

Comment:
In general simple string concatenation is less prone to errors from misplaced arguments than format strings.

Reply:
Okay, changed the occurrences of String.format() to connected strings. However, I seem to see the instance of String.format()  all over the places in the repo. There might be perf consideration to not have string concatenation which creates multiple string instances.

Reply:
This is a common example of cargo cult programming. Devs use format because they see others use it, and they think it must be somehow newer and better. It really isn't, though it does introduce a whole new class of bugs that don't appear with simple concatenation. There's no performance issue when the string is concatenated/formatted unconditionally as here in line 526. Either the code reaches line 526 and the string is constructed, or it doesn't get to that line and it isn't. The only case where format might be minimally faster is if the string is constructed before the VM knows if it will be used; e.g. in the argument list of Preconditions.checkArgument.